File: solutions/python/isbn-verifier/1/isbn_verifier.py
import re

def is_valid(isbn):
    if isbn == '':
        return False

    isbn = isbn.replace("-", "")
    isbn_verify = isbn[-1]
    digits = isbn[:-1]

    if (isbn_verify != 'X' and not isbn_verify.isdigit()) or re.search(r"[a-zA-Z]", digits):
        return False
    elif len(digits) == 9:
        numbers = [int(n) for n in digits]

        if isbn_verify == 'X':
            numbers.append(10)
        elif isbn_verify.isdigit():
            numbers.append(int(isbn_verify))
        else:
            return False

        total = sum(n * (10 - i) for i, n in enumerate(numbers))
        return total % 11 == 0
    else:
        return False

File: solutions/cpp/isbn-verifier/1/isbn_verifier.cpp
#include "isbn_verifier.h"

namespace isbn_verifier {

    bool is_valid(const std::string& isbn) {
        int sum = 0;
        int i = 0;
        for (const char c : isbn) {
            if ((c >= '0' && c <= '9') || c == 'X') {
                if (c == 'X' && i < 9) {
                    return false;
                }
                const int value = (c == 'X') ? 10 : c - '0';
                sum += value * (10 - i++);
            }else if (c == '-'){}
            else return false;
        }
        if ( i != 10) {
            return false;
        }
        return sum % 11 == 0;
    }

}  // namespace isbn_verifier
